packages/nuxt/src/core/nuxt.ts

```diff
@@ -347,13 +347,16 @@ async function initNuxt (nuxt: Nuxt) {
       return nuxt.callHook('restart', { hard: true })
     }
 
-    // User provided patterns
+    // User provided patterns:
+    // - nuxt.options.watch relative to srcDir
```

Comment:
we should also normalise the paths that are provided in the schema to absolute paths (`resolve(await get('srcDir'), val)`, which will make for a better solution for layers:

https://github.com/nuxt/nuxt/blob/03abb6aa4a879c2d3b97595cd3880c682c180dc2/packages/schema/src/config/common.ts#L373-L375

Reply:
The normalized check is required for Regex, for example, my playground config is (I have one.ts, the-two.ts and two.ts in root with dymmy content):
```ts
export default defineNuxtConfig({
  watch: ['one.ts', /^two/]
})
```

If we change `two.ts`, we must normalize it relative to `srcDir`, it will not match the regex.

Reply:
builder module is also normalizing the layers watch option to be relative to project `srcDir`, may we should revert my last commit:

https://github.com/nuxt/nuxt/blob/main/packages/nuxt/src/core/builder.ts#L98-L103

Reply:
I need to add a check, the path should be inside the layer `srcDir`